src/functions.c

```diff
@@ -370,6 +370,61 @@ void fcallCommandReadOnly(client *c) {
     fcallCommandGeneric(c, 1);
 }
 
+void functionsDumpCommand(client *c) {
+    unsigned char buf[2];
+    uint64_t crc;
+    rio payload;
+    rioInitWithBuffer(&payload,sdsempty());
+
+    functionsSaveRio(&payload);
+
+    /* RDB version */
+    buf[0] = RDB_VERSION & 0xff;
+    buf[1] = (RDB_VERSION >> 8) & 0xff;
+    payload.io.buffer.ptr = sdscatlen(payload.io.buffer.ptr,buf,2);
+
+    /* CRC64 */
+    crc = crc64(0,(unsigned char*)payload.io.buffer.ptr,
+                sdslen(payload.io.buffer.ptr));
+    memrev64ifbe(&crc);
+    payload.io.buffer.ptr = sdscatlen(payload.io.buffer.ptr,&crc,8);
+
+    addReplyBulkSds(c,payload.io.buffer.ptr);
+}
+
+void functionsRestoreCommand(client *c) {
+    uint16_t rdbver;
+    if (verifyDumpPayload(c->argv[2]->ptr, sdslen(c->argv[2]->ptr), &rdbver) != C_OK) {
+        addReplyError(c,"DUMP payload version or checksum are wrong");
+        return;
+    }
+
+    sds data = c->argv[2]->ptr;
+    size_t data_len = sdslen(data);
+    rio payload;
+    int type;
+
+    functionsCtx* f_ctx = functionsCtxCreate();
+    rioInitWithBuffer(&payload, data);
+
+    /* Read until reaching last 10 bytes that should contain RDB version and checksum. */
+    while(data_len - payload.io.buffer.pos > 10) {
+        /* Read type. */
+        if ((type = rdbLoadType(&payload)) == -1) goto load_error;
+        if (type != RDB_OPCODE_FUNCTION) goto load_error;
+        if (rdbFunctionLoad(&payload, rdbver, f_ctx) != C_OK) goto load_error;
+    }
+
+    functionsCtxSwapWithCurrent(f_ctx);
```

Comment:
maybe there's a use case for merging functions from two sources, or failing when a function already exists?
in which case we need a `REPLACE` argument, and to return an error when there's a name collision (of a function name or library name)?

Reply:
@oranagra maybe we should always do merge (and raise error on collision) and if the user wan REPLACE he can do `FUNCTION FLUSH` before sending the `FUNCTION RESTORE`?

Reply:
REPLACE (overriding some functions), is not the same as FLUSH + RESTORE.
i.e. since RESTORE carries multiple functions

Reply:
OK so you want the user to use `FUNCTION FLUSH` anyway in case he wants a full replacement of the functions set. And also allow the user to specify REPLACE argument to replace  collides functions, so: `FUNCTION RESTORE [REPLACE] <blob>`? Looks good?

Reply:
Yes, I suppose we don't need a MERGE
Argument, since the user can use FLUSH in MULTI? 
Still maybe it's more convenient to have some atomic option for MERGE or FLUSH

Reply:
@oranagra added FLUSH, APPEND, and REPLACE options according to this comment [here](https://github.com/redis/redis/pull/9938#discussion_r770276933)